src/main/java/com/eprosima/fastdds/idl/templates/TypeObjectTestingTestSource.stg

```diff
@@ -0,0 +1,1101 @@
+    ASSERT_EQ(bitfieldId, $bitset.name$_complete_bitfieldseq.size());
+}
+>>
+
+test_annotation(ctx, parent, annotation) ::= <<
+TEST(TypeObjectTests, TestAnnotationTypeObject_$annotation.name$)
+{
+    MemberId memberId;
+
+    //Minimal
+    $check_annotation_type(ctx=ctx, annotation=annotation, relation=false, relation_name="minimal", relation_capname="Minimal")$ 
+    memberId = 0;
+    $annotation.members:{ parameter | $check_annotation_parameter_type(ctx=ctx, parameter=parameter, relation=false, relation_name="minimal", relation_capname="Minimal")$ };separator="\n"$
+    ASSERT_EQ(memberId, $annotation.name$_minimal_memberseq.size());
+
+    //Complete
+    $check_annotation_type(ctx=ctx, annotation=annotation, relation=true, relation_name="complete", relation_capname="Complete")$ 
+    memberId = 0;
+    $annotation.members:{ parameter | $check_annotation_parameter_type(ctx=ctx, parameter=parameter, relation=true, relation_name="complete", relation_capname="Complete")$ };separator="\n"$
+    ASSERT_EQ(memberId, $annotation.name$_complete_memberseq.size());
+}
+>>
+
+test_const(ctx, parent, const) ::= <<
+TEST(TypeObjectTests, TestConstTypeObject_$const.name$)
+{
+    //Minimal
+    $check_const_type(ctx=ctx, const=const, relation=false, relation_name="minimal", relation_capname="Minimal")$ 
+
+    //Complete
+    $check_const_type(ctx=ctx, const=const, relation=true, relation_name="complete", relation_capname="Complete")$ 
+}
+>>
+
+
+fwd_decl(ctx, parent, type) ::= <<>>
+
+typedef_decl(ctx, parent, typedefs, typedefs_type, declarator_type) ::= <<
+$test_typedef(ctx=ctx, parent=parent, typedefs=typedefs)$
+>>
+
+struct_type(ctx, parent, struct, member_list) ::= <<
+$test_structure(ctx=ctx, parent=parent, struct=struct)$
+>>
+
+union_type(ctx, parent, union, switch_type) ::= <<
+$test_union(ctx=ctx, parent=parent, union=union)$
+>>
+
+enum_type(ctx, parent, enum) ::= <<
+$test_enum(ctx=ctx, parent=parent, enum=enum)$
+>>
+
+bitmask_type(ctx, parent, bitmask) ::= <<
+$test_bitmask(ctx=ctx, parent=parent, bitmask=bitmask)$
+>>
+
+bitset_type(ctx, parent, bitset) ::= <<
+$test_bitset(ctx=ctx, parent=parent, bitset=bitset)$
+>>
+
+annotation(ctx, parent, annotation) ::= <<
+$test_annotation(ctx=ctx, parent=parent, annotation=annotation)$
+>>
+
+module(ctx, parent, module, definition_list) ::= <<
+$definition_list$
+>>
+
+definition_list(definitions) ::= <<
+$definitions;separator="\n"$
+>>
+
+const_decl(ctx, parent, const) ::= <<
+$test_const(ctx=ctx, parent=parent, const=const)$
+>>
+
+
+member_type(ctx, member, type_member, declarators) ::= <<>>
+element_type(ctx, element, type_element, declarator) ::= <<>>
+sequence_type(ctx, sequence, type_sequence) ::= <<>>
+map_type(ctx, map, key_type, value_type) ::= <<>>
+string_type(ctx, string) ::= <<>>
+wide_string_type(ctx, wstring) ::= <<>>
+array_declarator(ctx, array, array_type) ::= <<>>
```

Comment:
This is not following the custom stg template. Please, fix.

Also, the sequence, map and array should call the corresponding tests.

Finally, the tests checking the string/wstring TypeIdentifiers are not included.